lib/internal/modules/cjs/loader.js

```diff
@@ -610,7 +623,8 @@ Module._findPath = function(request, paths, isMain) {
   }
 
   const cacheKey = request + '\x00' +
-                (paths.length === 1 ? paths[0] : paths.join('\x00'));
+                (paths.length === 1 ?
+                  paths[0] : ArrayPrototypeJoin(paths, '\x00'));
```

Comment:
this ternary seems like an optimization, otherwise it could just be `request + '\x00' + ArrayPrototypeJoin(paths, '\x00')`?

Reply:
I would think so. Maybe joining, even on a one-element array is slower than just returning the first entry. I am not sure about that though.